src/main/java/updater/UpdateManager.java

```diff
@@ -123,6 +157,63 @@ private boolean extractJarUpdater() {
     }
 
     /**
+     * Keeps the number of HT Jar in the folder used as backup to specified amount.
+     */
+    private void cleanupHTBackupJar() {
+        logger.info("Cleaning up backup JAR");
+
+        File currDirectory = new File(".");
+
+        File[] filesInCurrentDirectory = currDirectory.listFiles();
+
+        if (filesInCurrentDirectory == null) {
+            // current directory always exists
+            assert false;
+            return;
+        }
+        assert filesInCurrentDirectory != null;
+
+        List<File> listOfFilesInCurrDirectory = Arrays.asList(filesInCurrentDirectory);
+
+        List<File> allHtBackupFiles = listOfFilesInCurrDirectory.stream()
+                .filter(f -> !f.getName().equals(String.format("HubTurbo_%s.jar", Version.getCurrentVersion())) &&
+                             f.getName().matches(HT_BACKUP_FILENAME_PATTERN_STRING))
+                .sorted(getHtBackupFileComparatorByVersion())
+                .collect(Collectors.toList());
+
+        if (allHtBackupFiles.isEmpty()) {
+            return;
+        }
+
+        for (int i = 0; i < (allHtBackupFiles.size() - MAX_HT_BACKUP_JAR_KEPT); i++) {
+            logger.info("Deleting " + allHtBackupFiles.get(i).getName());
+            if (!allHtBackupFiles.get(i).delete()) {
+                logger.warn("Failed to delete old HT backup file " + allHtBackupFiles.get(i).getName());
+            }
+        }
+    }
+
+    private Comparator<File> getHtBackupFileComparatorByVersion() {
+        return (a, b) -> getVersionOfHtBackupFileFromFilename(a.getName())
+                .compareTo(getVersionOfHtBackupFileFromFilename(b.getName()));
+    }
+
+    /**
+     * Get version of HubTurbo from Jar backup file (filename is "HubTurbo_V[major].[minor].[patch].jar")
+     * @param filename filename of HT backup JAR
+     * @return version of HT of backup JAR, or Version 0 if filename not matching backup file
+     */
+    private Version getVersionOfHtBackupFileFromFilename(String filename) {
+        Pattern htJarFileBackupPattern = Pattern.compile(HT_BACKUP_FILENAME_PATTERN_STRING);
+        Matcher htJarFileBackupMatcher = htJarFileBackupPattern.matcher(filename);
+        if (!htJarFileBackupMatcher.find()) {
+            return new Version(0, 0, 0);
+        }
```

Comment:
If this should never happen (programmer error), I think it's fine to assert false. We can return version 0 in case, but that behaviour shouldn't be relied on. Maybe we can add to the comment also to say why version 0 is returned.